sim-lib/src/lib.rs

```diff
@@ -155,6 +157,15 @@ impl Display for PaymentResult {
     }
 }
 
+impl Default for PaymentResult {
+    fn default() -> Self {
+        PaymentResult {
+            htlc_count: 0,
+            payment_outcome: PaymentOutcome::Unknown,
+        }
+    }
+}
+
```

Comment:
I don't think we should implement `Default` here, but a constrictor called `unknown` or something along those lines. The main reasoning is that we are not using default here as it is normally intended, we're just calling `PaymentResult::default()` in some places of the codebase. Plus the default being an unknown result is a bit weird design-wise